WordPress/Classes/ViewRelated/Reader/Subscriptions/ReaderSubscriptionsView.swift

```diff
@@ -128,9 +129,66 @@ struct ReaderSubscriptionsView: View {
         ReaderSubscriptionHelper().unfollow(site)
     }
 
-    private func reloadSearchResults(searchText: String) {
-        let ranking = StringRankedSearch(searchTerm: searchText)
-        searchResults = ranking.search(in: subscriptions) { "\($0.title) \($0.siteURL)" }
+    private func performBackgroundSearch(searchText: String) {
+        struct SearchableSubscription: Sendable {
+            let objectID: NSManagedObjectID
+            let title: String
+            let siteURL: String
+
+            var searchableText: String {
+                "\(title) \(siteURL)"
+            }
+
+            init(_ subscription: ReaderSiteTopic) {
+                self.objectID = subscription.objectID
+                self.title = subscription.title
+                self.siteURL = subscription.siteURL
+            }
+        }
+
+        // Cancel any existing search task
+        searchTask?.cancel()
+
+        // Clear results immediately if search text is empty
+        if searchText.isEmpty {
+            searchResults = nil
+            pendingSearchText = nil
+            return
+        }
+
+        // Start new background search task
+        searchTask = Task {
+            // Store the search text we're processing
+            let currentSearchText = searchText
+
+            // Create searchable data on main thread to avoid Core Data context issues
+            let searchableData = subscriptions.map(SearchableSubscription.init)
+
+            // Perform the search on a background queue with parallel processing
+            let resultObjectIDs = await StringRankedSearch(searchTerm: currentSearchText)
+                .parallelSearch(in: searchableData) { $0.searchableText }
+                .map(\.objectID)
+
+            // Check if we were cancelled or if search text changed during search
+            guard !Task.isCancelled else { return }
+
+            // Update results on main thread
+            await MainActor.run {
+                // Only update if this search is still relevant
+                if currentSearchText == searchText {
+                    searchResults = subscriptions.filter { resultObjectIDs.contains($0.objectID) }
+                    pendingSearchText = nil
+                } else {
+                    // Search text changed during our search, mark that we need a new search
+                    pendingSearchText = searchText
+                }
+
+                // If there's a pending search, start it now
+                if let pendingSearchText {
+                    performBackgroundSearch(searchText: pendingSearchText)
+                }
```

Comment:
I find the use of `pendingSearchText` is a bit confusing. Every change to the search input would trigger a `performBackgroundSearch` call, which would perform a new search. That makes me think if this call here is redundant?